consensus/src/address.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::any::Any;
+
+use snarkvm_dpc::{testnet1::instantiated::Components, AccountAddress, AccountPrivateKey};
+
+pub trait AddressContainer: Any + Send + Sync {}
+
+// needs basedpccomponents to impl sync + send
+// impl<B: BaseDPCComponents> AddressContainer for AccountAddress<B> {
+
+// }
+
+impl AddressContainer for AccountAddress<Components> {}
+
+pub struct Address {
+    inner: Box<dyn Any + Send + Sync + 'static>,
+}
+
+impl<T: AddressContainer> From<T> for Address {
+    fn from(other: T) -> Self {
+        Address { inner: Box::new(other) }
+    }
+}
+
+impl Address {
+    pub fn into<T: AddressContainer>(self) -> T {
+        *self
+            .inner
+            .downcast()
+            .expect("illegal cross-network address contamination")
```

Comment:
Couldn't the node potentially receive a cross-network address from a malicious peer, causing a panic upon a downcast?

Reply:
it gets validated at deserialization from network, this is all stopgap for further snarkvm refactors or deeper multi-net support